### ISSUE TITLE
[Bug] When I select Bloody Palace, the game stops on the main screen and automatically ends.

### ISSUE THREAD

**Issue by GitHubro5** [muhopensores/dmc4_hook#9]
Hello.

When I select Bloody Palace, the game stops on the main screen and automatically ends to the Windows desktop.
When the game stops, all the game sounds stop, too.

I will attach the part where the game stops and write down the contents of dmc4_hook_log.

This issue is resolved by erasing the dinput8.dll file and does not occur in version DX10, with or without it.

All settings are using default values and unchecked.
I am using DMC4Hook 1.3 and game executable version 1.0.0.0.

As you know, there are two versions of Devil May Cry 4.
One is 1.0.0.0 and the other is 1.1.0.0.
I have two versions of Devil May Cry 4.
So I did some tests as follows.
I confirmed that DMC4Hook does not work at 1.1.0.0.
Additionally, I replaced the 1.0.0.0 executable with the 1.1.0.0 executable also vice versa and found that the game did not turn on in both cases.
Only the black screen will be displayed instead of the Capcom logo.

Besides, is it impossible to apply DMC4Hook to DX10 1.0.0.0 version executable file? and both DX9 1.1.0.0 version and DX10 1.1.0.0 version?

![Freezing Bloody Palace Main Screen](https://user-images.githubusercontent.com/14139205/130793640-034af9cf-681d-44bd-8941-0049db350bf5.png)

21:23:47 ======================================================
21:23:47                     LOG START                         
21:23:47 ======================================================
21:23:47 
21:23:47 dinput8.dll build date 
, commit hash #======================================================

21:23:47 
21:23:47 PassiveEnemies::onInitialize()
21:23:47 FastStart::onInitialize()
21:23:47 [FastStart]: Installing hook at 008DB77D.
21:23:47 EasyJc::onInitialize()
21:23:47 [EasyJc]: Installing hook at 004AB336.
21:23:47 AreaJump::onInitialize()
21:23:47 [AreaJump]: Installing hook at 0040E1F6.
21:23:47 LimitAdjust::onInitialize()
21:23:47 InfAirHikes::onInitialize()
21:23:47 NoHbKnockback::onInitialize()
21:23:47 [NoHbKnockback]: Installing hook at 0051C384.
21:23:47 MoveIds::onInitialize()
21:23:47 [MoveIds]: Installing hook at 0083EBD6.
21:23:47 SelectiveCancels::onInitialize()
21:23:47 [SelectiveCancels]: Installing hook at 0080332A.
21:23:47 CharacterSwap::onInitialize()
21:23:47 WorkRate::onInitialize()
21:23:47 [WorkRate]: Installing hook at 0040A948.
21:23:47 InfPlayerHealth::onInitialize()
21:23:47 [InfPlayerHealth]: Installing hook at 004FF282.
21:23:47 BackgroundRendering::onInitialize()
21:23:47 ERROR: hWindow is null!
21:23:47 [BackgroundRendering]: CreateHookApi(user32, GetForegroundWindow) returned MH_OK
21:23:47 [BackgroundRendering]: EnableHook(&GetForegroundWindow) returned MH_OK
21:23:47 [BackgroundRendering]: CreateHookApi(user32, GetFocus) returned MH_OK
21:23:48 [BackgroundRendering]: EnableHook(&GetFocus) returned MH_OK
21:23:48 DisableDarkslayer::onInitialize()
21:23:48 EnemyDT::onInitialize()
21:23:48 InfiniteTime::onInitialize()
21:23:48 Borderless::onInitialize()
21:23:48 InfDreadnought::onInitialize()
21:23:48 [InfDreadnought]: Installing hook at 007BDB76.
21:23:48 NoClip::onInitialize()
21:23:48 ManualTwosomeTime::onInitialize()
21:23:48 SkipPandora::onInitialize()
21:23:48 [SkipPandora]: Installing hook at 007B122F.
21:23:48 PlayerTracker::onInitialize()
21:23:48 [PlayerTracker]: Installing hook at 007A8337.
21:23:48 [PlayerTracker]: Installing hook at 007A838C.
21:23:48 RestoreMaxHp::onInitialize()
21:23:48 [RestoreMaxHp]: Installing hook at 00543BC1.
21:23:48 [RestoreMaxHp]: Installing hook at 005A6042.
21:23:48 [RestoreMaxHp]: Installing hook at 005653F5.
21:23:48 [RestoreMaxHp]: Installing hook at 00583ACF.
21:23:48 [RestoreMaxHp]: Installing hook at 0060DFB6.
21:23:48 [RestoreMaxHp]: Installing hook at 005F902F.
21:23:48 [RestoreMaxHp]: Installing hook at 005E08D8.
21:23:48 [RestoreMaxHp]: Installing hook at 005B6FDC.
21:23:48 [RestoreMaxHp]: Installing hook at 005D3603.
21:23:48 [RestoreMaxHp]: Installing hook at 0061D971.
21:23:48 [RestoreMaxHp]: Installing hook at 00635FF9.
21:23:48 [RestoreMaxHp]: Installing hook at 0064E6A3.
21:23:48 [RestoreMaxHp]: Installing hook at 0068A431.
21:23:48 [RestoreMaxHp]: Installing hook at 006ACE60.
21:23:48 [RestoreMaxHp]: Installing hook at 006C1D77.
21:23:48 [RestoreMaxHp]: Installing hook at 00705702.
21:23:48 [RestoreMaxHp]: Installing hook at 004BE16E.
21:23:48 [RestoreMaxHp]: Installing hook at 00494D60.
21:23:48 EnemySlotting::onInitialize()
21:23:48 [EnemySlotting]: Installing hook at 0073771B.
21:23:48 InputStates::onInitialize()
21:23:48 [InputStates]: Installing hook at 007B0847.
21:23:48 FpsLimit::onInitialize()
21:23:48 [FpsLimit]: Installing hook at 0090D89C.
21:23:48 MoveIdsNero::onInitialize()
21:23:48 [MoveIdsNero]: Installing hook at 008212B6.
21:23:48 TimerMem::onInitialize()
21:23:48 [TimerMem]: Installing hook at 007AD76A.
21:23:48 [TimerMem]: Installing hook at 00805BC1.
21:23:48 InfSkyStars::onInitialize()
21:23:48 FreeJc::onInitialize()
21:23:48 ForceLucifer::onInitialize()
21:23:48 [ForceLucifer]: Installing hook at 00831CF0.
21:23:48 FastSprint::onInitialize()
21:23:48 LdkWithDmd::onInitialize()
21:23:48 [LdkWithDmd]: Installing hook at 00494AED.
21:23:48 FreezeEnemies::onInitialize()
21:23:48 InfAllHealth::onInitialize()
21:23:48 DmdBloodyPalace::onInitialize()
21:23:48 NoDeath::onInitialize()
21:23:48 DmdLevelAi::onInitialize()
21:23:48 RemoveLaunchArmour::onInitialize()
21:23:48 OneHitKill::onInitialize()
21:23:48 CameraSettings::onInitialize()
21:23:48 [CameraSettings]: Installing hook at 004191C5.
21:23:48 [CameraSettings]: Installing hook at 0041946C.
21:23:48 [CameraSettings]: Installing hook at 0041A140.
21:23:48 [CameraSettings]: Installing hook at 0041914C.
21:23:48 [CameraSettings]: Installing hook at 004198E6.
21:23:48 [CameraSettings]: Installing hook at 004180EA.
21:23:48 [CameraSettings]: Installing hook at 00418193.
21:23:48 [CameraSettings]: Installing hook at 00422D5F.
21:23:48 [CameraSettings]: Installing hook at 0042261A.
21:23:48 HeightRestrictionDante::onInitialize()
21:23:48 InfTrickRange::onInitialize()
21:23:48 InfFaustCloak::onInitialize()
21:23:48 HeightRestrictionNero::onInitialize()
21:23:48 CutsceneSkip::onInitialize()
21:23:48 EnemySpawn::onInitialize()
21:23:48 KnockbackEdits::onInitialize()
21:23:48 DisableCameraEvents::onInitialize()
21:23:48 NoDtCooldown::onInitialize()
21:23:48 DtEnemiesDontStun::onInitialize()
21:23:48 TrackingFullHouse::onInitialize()
21:23:48 [TrackingFullHouse]: Installing hook at 007D3471.
21:23:48 RoseRemovesPins::onInitialize()
21:23:48 [RoseRemovesPins]: Installing hook at 008158C3.
21:23:48 TrickDown::onInitialize()
21:23:48 [TrickDown]: Installing hook at 007CB119.
21:23:48 [TrickDown]: Installing hook at 007CB33D.
21:23:48 DamageMultiplier::onInitialize()
21:23:48 [DamageMultiplier]: Installing hook at 0051BFE5.
21:23:48 HpInOrbsDisplay::onInitialize()
21:23:48 [HpInOrbsDisplay]: Installing hook at 004FDD35.
21:23:48 InfDT::onInitialize()
21:23:48 [InfDT]: Installing hook at 004FF315.
21:23:48 BerialDaze::onInitialize()
21:23:48 [BerialDaze]: Installing hook at 0064271A.
21:23:48 StunAnything::onInitialize()
21:23:48 BpPortal::onInitialize()
21:23:48 HideHud::onInitialize()
21:23:48 FastPandora::onInitialize()
21:23:48 BpBossRush::onInitialize()
21:23:48 InfRevive::onInitialize()
21:23:48 InfTableHopper::onInitialize()
21:23:48 EnemyAttackOffScreen::onInitialize()
21:23:48 SlowWalk::onInitialize()
21:23:48 DifficultySelect::onInitialize()
21:23:48 EnemyReplace::onInitialize()
21:23:48 BpJumpHook::onInitialize()
21:23:48 [BpJumpHook]: Installing hook at 008AB8FA.
21:23:48 [BpJumpHook]: Installing hook at 0044A974.
21:23:48 NoAutomaticCharacters::onInitialize()
21:23:48 [NoAutomaticCharacters]: Installing hook at 00778E75.
21:23:48 [NoAutomaticCharacters]: Installing hook at 00779082.
21:23:48 DisableChimeraBlades::onInitialize()
21:23:48 [DisableChimeraBlades]: Installing hook at 005F1A8F.
21:23:48 Loading config from dmc4_hook.cfg
21:23:48 PassiveEnemies::onConfigLoad()
21:23:48 FastStart::onConfigLoad()
21:23:48 EasyJc::onConfigLoad()
21:23:48 [EasyJc]: Installing patch at 00827999.
21:23:48 AreaJump::onConfigLoad()
21:23:48 LimitAdjust::onConfigLoad()
21:23:48 InfAirHikes::onConfigLoad()
21:23:48 NoHbKnockback::onConfigLoad()
21:23:48 MoveIds::onConfigLoad()
21:23:48 SelectiveCancels::onConfigLoad()
21:23:48 CharacterSwap::onConfigLoad()
21:23:48 WorkRate::onConfigLoad()
21:23:48 InfPlayerHealth::onConfigLoad()
21:23:48 BackgroundRendering::onConfigLoad()
21:23:48 DisableDarkslayer::onConfigLoad()
21:23:48 EnemyDT::onConfigLoad()
21:23:48 InfiniteTime::onConfigLoad()
21:23:48 Borderless::onConfigLoad()
21:23:48 InfDreadnought::onConfigLoad()
21:23:48 NoClip::onConfigLoad()
21:23:48 ManualTwosomeTime::onConfigLoad()
21:23:48 SkipPandora::onConfigLoad()
21:23:48 PlayerTracker::onConfigLoad()
21:23:48 RestoreMaxHp::onConfigLoad()
21:23:48 EnemySlotting::onConfigLoad()
21:23:48 InputStates::onConfigLoad()
21:23:48 FpsLimit::onConfigLoad()
21:23:48 MoveIdsNero::onConfigLoad()
21:23:48 TimerMem::onConfigLoad()
21:23:48 InfSkyStars::onConfigLoad()
21:23:48 FreeJc::onConfigLoad()
21:23:48 [FreeJc]: Installing patch at 00827999.
21:23:48 ForceLucifer::onConfigLoad()
21:23:48 FastSprint::onConfigLoad()
21:23:48 LdkWithDmd::onConfigLoad()
21:23:48 FreezeEnemies::onConfigLoad()
21:23:48 InfAllHealth::onConfigLoad()
21:23:48 DmdBloodyPalace::onConfigLoad()
21:23:48 NoDeath::onConfigLoad()
21:23:48 DmdLevelAi::onConfigLoad()
21:23:48 RemoveLaunchArmour::onConfigLoad()
21:23:48 OneHitKill::onConfigLoad()
21:23:48 CameraSettings::onConfigLoad()
21:23:48 HeightRestrictionDante::onConfigLoad()
21:23:48 InfTrickRange::onConfigLoad()
21:23:48 InfFaustCloak::onConfigLoad()
21:23:48 HeightRestrictionNero::onConfigLoad()
21:23:48 CutsceneSkip::onConfigLoad()
21:23:48 EnemySpawn::onConfigLoad()
21:23:48 KnockbackEdits::onConfigLoad()
21:23:48 DisableCameraEvents::onConfigLoad()
21:23:48 NoDtCooldown::onConfigLoad()
21:23:48 DtEnemiesDontStun::onConfigLoad()
21:23:48 TrackingFullHouse::onConfigLoad()
21:23:48 RoseRemovesPins::onConfigLoad()
21:23:48 TrickDown::onConfigLoad()
21:23:48 DamageMultiplier::onConfigLoad()
21:23:48 HpInOrbsDisplay::onConfigLoad()
21:23:48 InfDT::onConfigLoad()
21:23:48 BerialDaze::onConfigLoad()
21:23:48 StunAnything::onConfigLoad()
21:23:48 BpPortal::onConfigLoad()
21:23:48 HideHud::onConfigLoad()
21:23:48 FastPandora::onConfigLoad()
21:23:48 BpBossRush::onConfigLoad()
21:23:48 InfRevive::onConfigLoad()
21:23:48 InfTableHopper::onConfigLoad()
21:23:48 EnemyAttackOffScreen::onConfigLoad()
21:23:48 SlowWalk::onConfigLoad()
21:23:48 DifficultySelect::onConfigLoad()
21:23:48 EnemyReplace::onConfigLoad()
21:23:48 BpJumpHook::onConfigLoad()
21:23:48 NoAutomaticCharacters::onConfigLoad()
21:23:48 DisableChimeraBlades::onConfigLoad()
21:23:49 [D3D Device present] D3DDEVICE_CREATION_PARAMETERS hFocusWindow=60794
21:23:49 Setting window hook, param = 60794 hWindow = 0

**Comment by GitHubro5** [comment]
Thank you for your answer.

I tested it with the file you attached and found that it worked well.

I couldn't find the steam drm free version, but I got many iso files with multiple torrents and tested the cracked exe files that exist in each of them.
However, the same problem has all occurred.

I checked that I can enter Bloody Palace even when I activate the function Skip Pandora.
But I don't know exactly what role this function plays yet.

Thank you.
